config/routes.rb

```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+  root to: 'recipes#index'
+
+  devise_for :users
+
+  get 'public_recipes', to: 'public_recipes#index', as: :public_recipes
+
+  resources :recipes do
+    resources :foods, only: [:new, :create]
+    resources :recipe_foods, only: [:new, :create, :index, :destroy]
+    member do
+      patch 'toggle_public'
+    end
+  end
+
+  resources :shopping_lists
```

Comment:
[REQUIRED] I noticed that you have only an index page for your shopping list but you are creating all routes for it. Kindly create only necessary routes and save resources